dex-it/src/test/scala/com/wavesplatform/it/matcher/api/http/GetOrderStatusInfoByIdWithSignature.scala

```diff
@@ -0,0 +1,204 @@
+          Status.Filled.name,
+          wavesUsdPair,
+          2.usd,
+          o.version,
+          20.usd
+        ))
+      }
+
+      withClue(" - cancelled") {
+        val o = mkOrder(alice, wavesUsdPair, BUY, 10.waves, 2.usd)
+        placeAndAwaitAtDex(o)
+        cancelAndAwait(alice, o)
+        validate200Json(dex1.rawApi.getOrderStatusInfoByIdWithSignature(alice, o)) should be(HttpOrderBookHistoryItem(
+          o.id(),
+          BUY,
+          AcceptedOrderType.Limit,
+          10.waves,
+          0,
+          2.usd,
+          0.003.waves,
+          0,
+          o.feeAsset,
+          o.timestamp,
+          Status.Cancelled.name,
+          wavesUsdPair,
+          0,
+          o.version,
+          0
+        ))
+      }
+    }
+
+    "should return error when the order doesn't exist" in {
+      val order = mkOrder(alice, wavesUsdPair, BUY, 10.waves, 2.usd)
+      validateMatcherError(
+        dex1.rawApi.getOrderStatusInfoByIdWithSignature(alice, order),
+        StatusCodes.NotFound,
+        9437193,
+        s"The order ${order.idStr()} not found"
+      )
+    }
+
+    "should return error exception when the publicKey is not correct base58 string" in {
+      val ts = System.currentTimeMillis
+      val sign = Base58.encode(crypto.sign(alice, alice.publicKey ++ Longs.toByteArray(ts)))
+
+      validate404Exception(dex1.rawApi.getOrderStatusInfoByIdWithSignature("null", "null", ts, sign))
+    }
+
+    "should return error exception when the orderId is not correct base58 string" in {
+      val order = mkOrder(alice, wavesUsdPair, BUY, 10.waves, 2.usd)
+      placeAndAwaitAtDex(order)
+      val ts = System.currentTimeMillis
+      val sign = Base58.encode(crypto.sign(alice, alice.publicKey ++ Longs.toByteArray(ts)))
+
+      validate404Exception(dex1.rawApi.getOrderStatusInfoByIdWithSignature(Base58.encode(alice.publicKey), "null", ts, sign))
+    }
+
+    "should return error if publicKey parameter has the different value of used in signature" in {
+      val order = mkOrder(alice, wavesUsdPair, BUY, 10.waves, 2.usd)
+      placeAndAwaitAtDex(order)
+      val ts = System.currentTimeMillis
+      val sign = Base58.encode(crypto.sign(alice, alice.publicKey ++ Longs.toByteArray(ts)))
+
+      validateIncorrectSignature(dex1.rawApi.getOrderStatusInfoByIdWithSignature(Base58.encode(bob.publicKey), order.idStr(), ts, sign))
+    }
+
+    "should return error if timestamp header has the different value of used in signature" in {
+      val order = mkOrder(alice, wavesUsdPair, BUY, 10.waves, 2.usd)
+      placeAndAwaitAtDex(order)
+      val ts = System.currentTimeMillis
+      val sign = Base58.encode(crypto.sign(alice, alice.publicKey ++ Longs.toByteArray(ts)))
+
+      validateIncorrectSignature(dex1.rawApi.getOrderStatusInfoByIdWithSignature(Base58.encode(alice.publicKey), order.idStr(), ts + 1000, sign))
+    }
+
+    // There is an incorrect error (Asset not found), we should discuss about it
```

Comment:
Have no idea how we can fix this right now without breaking a backward compatibility. I created a separate task for it: DEX-982. You could update TODO here ;)